ArduPlane/quadplane.cpp

```diff
@@ -2267,50 +2288,109 @@ void QuadPlane::vtol_position_controller(void)
     // horizontal position control
     switch (poscontrol.state) {
 
+    case QPOS_APPROACH:
+    case QPOS_AIRBRAKE: {
+        /*
+          fixed wing approach to point where we need to transition to VTOL
+         */
+        const Vector2f diff_wp = plane.current_loc.get_distance_NE(loc);
+        const float distance = diff_wp.length();
+        const float groundspeed = ahrs.groundspeed();
+
+        // run fixed wing navigation
+        plane.nav_controller->update_waypoint(plane.prev_WP_loc, loc);
+
+        // use TECS for throttle
+        SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, plane.SpdHgt_Controller->get_throttle_demand());
+
+        // use TECS for pitch
+        int32_t commanded_pitch = plane.SpdHgt_Controller->get_pitch_demand();
+        plane.nav_pitch_cd = constrain_int32(commanded_pitch, plane.pitch_limit_min_cd, plane.aparm.pitch_limit_max_cd.get());
+
+        // use nav controller roll
+        plane.calc_nav_roll();
+
+        // see if we have reached the transition point
+        uint16_t qrtl_radius = abs(plane.g.rtl_radius);
+        if (qrtl_radius == 0) {
+            qrtl_radius = abs(plane.aparm.loiter_radius);
+        }
+        const float stop_distance = MAX(qrtl_radius, stopping_distance());
+
+        /*
+          see if we should start airbraking stage. For non-tailsitters
+          we can use the VTOL motors as airbrakes by firing them up
+          before we transition. This gives a smoother transition and
+          gives us a nice lot of deceleration
+         */
+        const float airbrake_time = 2;
+        if (!is_tailsitter() &&
+             poscontrol.state == QPOS_APPROACH &&
+            distance < stop_distance + airbrake_time*groundspeed) {
```

Comment:
it falls thru to the normal stopping distance test as well, so should go straight to QPOS_POSITION1 if we start this distance <= stopping distance
I agree though that a helper would be good, and would allow us to use airbrake in the circle-to-land code and QRTL